ebu_tt_live/documents/ebutt3_deduplicator.py

```diff
@@ -1,20 +1,68 @@
 from datetime import timedelta
-import logging
+from logging import getLogger
 import copy
 from pyxb.binding.basis import NonElementContent, ElementContent
 from ebu_tt_live.bindings import tt
+from ..bindings.validation.base import SemanticValidationMixin
+from ..bindings.pyxb_utils import RecursiveOperation
 
 
 log = logging.getLogger(__name__)
 
 
 class EBUTT3Deduplicator(object):
 
-    _style_segment = None
-    _style_list = []
-    _new_label = None
-    _sequence_number = None
+    _style_id = None
+    _style_attribute = None
     _sequence_identifier = None
+    _sequence_number = None
+
+    def __init__(self, style_id, style_attribute, sequence_identifier, sequence_number):
+        if style_id is None or style_attribute is None:
+            raise Exception()
+
+        self._style_id = list(style_id)
+        self._style_attribute = list(style_attribute)
+        self._sequence_idtentifier = sequence_identifier
+        self._sequence_number = sequence_number
+        self._perform_deduplication()
```

Comment:
I get what your trying todo but i don't think this is the best approach. Usually the init is to just setup the class but not start execution.